src/components/TodoForm/TodoForm.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { useState } from 'react';
+
+import usersFromServer from '../../api/users';
+import { getUserById } from '../../services/getUsers';
+import { getMaxNumber } from '../../services/getMaxNumber';
+import { Todo } from '../../types/todo';
+
+type Props = {
+  onSubmit: (todo: Todo) => void;
+  todos: Todo[];
+};
+
+export const TodoForm: React.FC<Props> = ({ onSubmit, todos }) => {
+  const [title, setTitle] = useState('');
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [userId, setUserId] = useState(0);
+  const [hasUserIdError, setHasUserIdError] = useState(false);
+
+  const handlTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setHasTitleError(false);
+  };
+
+  const handlUserIdChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setUserId(+event.target.value);
+    setHasUserIdError(false);
+  };
+
+  const reset = () => {
+    setTitle('');
+    setUserId(0);
+    setHasTitleError(false);
+    setHasUserIdError(false);
+  };
+
+  const handlSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    setHasTitleError(!title);
+    setHasUserIdError(!userId);
+
+    if (!title || !userId) {
+      return;
+    }
+
+    onSubmit({
+      id: getMaxNumber(todos),
+      title: title.trim(),
+      completed: false,
+      userId: userId,
+      user: getUserById(userId),
```

Comment:
[CHECKLIST - Prepare data in one place] - You are calling the 'getUserById' function inside the 'onSubmit' function. This is not recommended. You should prepare all data in the App component and pass it down to child components as props.